Description: Disable jemalloc tests and on s390x
Author: Ximin Luo <infinity0@debian.org>
Bug: https://github.com/rust-lang/rust/issues/38596
Bug: https://github.com/rust-lang/rust/issues/38612
Bug: https://github.com/rust-lang/rust/pull/38675
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/src/librustc_back/target/s390x_unknown_linux_gnu.rs
+++ b/src/librustc_back/target/s390x_unknown_linux_gnu.rs
@@ -19,6 +19,8 @@
     // Pass the -vector feature string to LLVM to respect this assumption.
     base.features = "-vector".to_string();
     base.max_atomic_width = Some(64);
+    // see #36994
+    base.exe_allocation_crate = "alloc_system".to_string();
 
     Ok(Target {
         llvm_target: "s390x-unknown-linux-gnu".to_string(),
--- a/src/test/compile-fail/allocator-rust-dylib-is-jemalloc.rs
+++ b/src/test/compile-fail/allocator-rust-dylib-is-jemalloc.rs
@@ -29,9 +29,9 @@
 // ensure we get the same error.
 //
 // So long as we CI linux/OSX we should be good.
-#[cfg(any(target_os = "linux", target_os = "macos"))]
+#[cfg(any(all(target_os = "linux", not(any(target_arch = "aarch64", target_arch = "mips", target_arch = "mips64", target_arch = "powerpc", target_arch = "powerpc64", target_arch = "s390x"))), target_os = "macos"))]
 extern crate alloc_system;
-#[cfg(not(any(target_os = "linux", target_os = "macos")))]
+#[cfg(not(any(all(target_os = "linux", not(any(target_arch = "aarch64", target_arch = "mips", target_arch = "mips64", target_arch = "powerpc", target_arch = "powerpc64", target_arch = "s390x"))), target_os = "macos")))]
 extern crate allocator1;
 
 fn main() {
--- a/src/test/run-pass/allocator-default.rs
+++ b/src/test/run-pass/allocator-default.rs
@@ -10,7 +10,7 @@
 
 #![feature(alloc_jemalloc)]
 
-#[cfg(any(target_os = "linux", target_os = "macos"))]
+#[cfg(any(all(target_os = "linux", not(any(target_arch = "aarch64", target_arch = "mips", target_arch = "mips64", target_arch = "powerpc", target_arch = "powerpc64", target_arch = "s390x"))), target_os = "macos"))]
 extern crate alloc_jemalloc;
 
 fn main() {
